src/assembler.rs

```diff
@@ -399,6 +399,120 @@ fn assemble_constructor(constructor: Function, fields: &[ContractField]) -> Toke
     output
 }
 
+/// Iter by every statement in `statements` and find cases, when `Statement::Comment(_)`
+/// is between if/else, if/else if, try/catch or catch/catch blocks.
+///
+/// Moves this comment to next block start.
+///
+/// Returns fixed vector of statements.
+fn check_statements(statements: &Vec<Statement>) -> Vec<Statement> {
```

Comment:
I think it would be better if we did this check in parse_statements in parser

parse a statement, if it is one of the statements you are looking for here (catch, else, else if, etc.) and the previous is a comment, then switch them